[Jensenkuok] iP

Jensenkuok opened the pull request.

Commit:
Add Increment: Level-0

Commit:
Add increment: Level-1 (Echo)

Commit:
Add increment: Level-2 (Add, List)

Commit:
Add increment: Level 3 (Mark as Done)

Commit:
A-CodingStandard

Commit:
Changed "yo" to "hey man" based on lecture recommendation

Commit:
Added ToDo class and updated main

Commit:
Added Deadline class and updated main

Changed the language that Jamal uses for better coherence

Commit:
Added Event class and updated main

Cheng-Zhiyuan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+    @Override
+    public String getTaskDisplay() {
+        return "[E]" + super.getTaskDisplay() + " (from: " + from + " to: " + to + ")";
+    }
```
Great camelCase!

Cheng-Zhiyuan commented on src/main/java/Jamal.java:
```diff
@@ -0,0 +1,155 @@
+
+                    String taskDescription = input.substring(6, fromIndex).trim();
+                    String from = input.substring(fromIndex + 6, toIndex).trim();
+                    String to = input.substring(toIndex + 4).trim();
```
Perhaps make the numbers into a constant?

Cheng-Zhiyuan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public void unmarkAsDone() {
+        this.isDone = false;
```
Great naming convention!

Cheng-Zhiyuan commented on src/main/java/Jamal.java:
```diff
@@ -0,0 +1,155 @@
+                try {
+                    int taskIndex = Integer.parseInt(input.substring(5)) - 1;
+                    if (taskIndex >= 0 && taskIndex < taskCount) {
+                        tasks[taskIndex].markAsDone();
```
Great structure of the code!

Cheng-Zhiyuan reviewed (commented):
Overall very clear and easy to read!

Teng-Yong-Quan commented on src/main/java/Jamal.java:
```diff
@@ -0,0 +1,157 @@
+                + "      O                         O \n"
+                + "      o                         o \n"
+                + "      O                         O \n"
+                + "      o  .oOoO' `oOOoOO. .oOoO' o \n"
```
It would be better to extract the string logo and welcome message into the method, so the number of lines in the main method could be reduced.

Teng-Yong-Quan commented on src/main/java/Jamal.java:
```diff
@@ -0,0 +1,157 @@
+            } else if (input.startsWith("unmark ")) {
+                try {
+                    int taskIndex = Integer.parseInt(input.substring(7)) - 1;
+                    if (taskIndex >= 0 && taskIndex < taskCount) {
```
I have noticed multiple magic numbers. Maybe you can avoid using them or assigning these numbers with variable names?

Teng-Yong-Quan commented on src/main/java/Jamal.java:
I like that there are no arrowhead style code in this file.

Teng-Yong-Quan commented on src/main/java/Jamal.java:
I think the codes in the main method could be segmented into different methods for better readability and to minimise the number of lines of code for each 'if-else' condition.

Commit:
Refactored code to use MAX_TASKS constant instead of hardcoded value 100 for task limit

Commit:
Extract welcome message and logo into JamalUI class for better readability

Commit:
Added more methods to JamalUI class

Commit:
Added try-catch required by weekly task

Commit:
Merge branch 'branch-Level-5'

Commit:
Organised into jamal package

Commit:
Merge branch 'branch-A-Packages'

Commit:
Used ArrayList to store tasks instead of task array

Commit:
Added TaskManager class
Refactor main to contain less implementation

Commit:
Added delete method to delete tasks

Commit:
Added storage class to handle storage

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

aditichadha1310 commented on src/main/java/jamal/Jamal.java:
```diff
@@ -0,0 +1,109 @@
+                    JamalUI.showSeparator();
+                }
+            }
+        }
```
Try to avoid such long methods. Keep the methods to about 30 Lines of code.

aditichadha1310 commented on src/main/java/jamal/Storage.java:
```diff
@@ -0,0 +1,101 @@
+            return type + " | " + status + " | " + description + " | " + ((Deadline) task).by;
+        } else if (task instanceof Event) {
+            type = "E";
+            return type + " | " + status + " | " + description + " | " + ((Event) task).from + " to " + ((Event) task).to;
```
Avoid magic strings

aditichadha1310 reviewed (commented):
overall. good job!

Commit:
Refactored code to organise packages & add getter methods

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Added Command Class and Parser to demonstrate better OOP

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Added a FindCommand class to allow users to find keyword specific tasks
Added empty line at the of files

Commit:
Level 8. Dates and Times: Jamal now recognises dates and times for deadline tasks

Commit:
Added header comments to all non-private classes/methods, and non-trivial private methods.

Jensenkuok committed:
Merge pull request #1 from Jensenkuok/branch-level-8

Branch level 8

Commit:
Jamal now understand dates and times

Jensenkuok committed:
Merge pull request #2 from Jensenkuok/branch-A-JavaDoc

Added header comments to all non-private classes/methods

Commit:
Resolved merge conflicts between master and JavaDoc branch

# Conflicts:
#	data/duke.txt
#	src/main/java/jamal/logic/Parser.java
#	src/main/java/jamal/tasks/Deadline.java

Commit:
Merged A-JavaDoc branch with Master

Commit:
Editted JavaDoc and changed textfile from duke.txt to jamal.txt

Commit:
Updated project README

Commit:
Updated project README

Commit:
Rectified spelling error in README
